src/internal.jl

```diff
@@ -13,26 +13,37 @@ end
 serialize(val::SystemUnitsSettings) = serialize_struct(val)
 deserialize(T::Type{<:SystemUnitsSettings}, val::Dict) = deserialize_struct(T, val)
 
+@kwdef mutable struct SharedSystemReferences <: InfrastructureSystemsType
+    supplemental_attribute_manager::Any = nothing
+    time_series_manager::Any = nothing
+end
+
 """
 Internal storage common to InfrastructureSystems types.
 """
 mutable struct InfrastructureSystemsInternal <: InfrastructureSystemsType
     uuid::Base.UUID
+    shared_system_references::Union{Nothing, SharedSystemReferences}
     units_info::Union{Nothing, UnitsData}
     ext::Union{Nothing, Dict{String, Any}}
 end
 
 """
 Creates InfrastructureSystemsInternal with a new UUID.
 """
-InfrastructureSystemsInternal(; uuid = make_uuid(), units_info = nothing, ext = nothing) =
-    InfrastructureSystemsInternal(uuid, units_info, ext)
+InfrastructureSystemsInternal(;
+    uuid = make_uuid(),
+    shared_system_references = nothing,
+    units_info = nothing,
+    ext = nothing,
+) =
+    InfrastructureSystemsInternal(uuid, shared_system_references, units_info, ext)
 
 """
 Creates InfrastructureSystemsInternal with an existing UUID.
 """
-InfrastructureSystemsInternal(u::UUIDs.UUID) =
-    InfrastructureSystemsInternal(u, nothing, nothing)
+InfrastructureSystemsInternal(u::Base.UUID) =
```

Comment:
I changed it because it made the Julia language server happy (it was complaining, which must be a bug). Ultimately, it doesn't matter.
```
julia> UUIDs.UUID === Base.UUID
true
```